### PR TITLE
Meet smoothy

### PR DESCRIPTION
**Pull request by AljoschaMeyer**
I have no idea what I am doing. It felt appropriate not to change anything about that until after submitting the PR.

How do you feel about adding dependencies? Using a VecDeque rather than a fixed-size ringbuffer hurts a bit.

### CODE REVIEW

File: src/smoothy.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Disclaimer:
+// - Beyond an educated guess that samples might denote an amplitude between -1 to 1, I have no idea how any of this works.
```
Comment (sandreae) [inline src/smoothy.rs:2]: This is how it works ;-p

File: src/smoothy.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// - I have not listened to what this produces. The initial idea was to make things sluggish by limiting how much samples would differ from the previous ones. But perhaps all this does is to reduce volume in a slightly random way? Only one way to find out: submitting a PR and having you describe the effect to me.
```
Comment (sandreae) [inline src/smoothy.rs:3]: I found with lower "blend" values it sounds like a high pass filter, once the blend values get higher then it brings the volume down.

**Comment by sandreae** [comment]
Hey Aljoscha! Thanks for the new module :-)

I tried it out and it's great :+1: 

> How do you feel about adding dependencies? Using a VecDeque rather than a fixed-size ringbuffer hurts a bit.

I'd like to keep dependencies down, but your suggestion seems generally useful so I'd be happy for that to be added in.

**Comment by sandreae** [comment]
Could you add a line to https://github.com/sandreae/earwig#modules describing the module a little?

**Comment by AljoschaMeyer** [comment]
I took the liberty of organizing the module list a bit, as well as asking contributors to include module information in the readme.

**Comment by sandreae** [comment]
Brilliant thanks!

**Pull request merged**